src/components/Swap.tsx

```diff
@@ -239,38 +244,14 @@ const Swap = ({ transferChains }: SwapProps) => {
           toAddress: '',
           slippage: optionSlippage / 100,
         }
-
-        // cancel previously running requests
-        if (source) {
-          source.cancel('cancel for new request')
-        }
-        source = axios.CancelToken.source()
-        const cancelToken = source.token
-        const config = {
-          cancelToken,
-        }
-
         try {
-          const result = await axios.post<any>(
-            process.env.REACT_APP_API_URL + 'transfer',
-            { deposit, withdraw },
-            config,
-          )
-          // filter if needed
-          const routes: Array<Array<TransferStep>> = result.data
-          setRoutes(routes)
-          fadeInAnimation(routeCards)
-          setHighlightedIndex(routes.length === 0 ? -1 : 0)
-          setNoRoutesAvailable(routes.length === 0)
+          const id = uuid()
+          setCurrentRouteCallId(id)
+          const result = await LIFI.findRoutes(deposit, withdraw)
+          setRouteCallResult({ result, id })
+        } catch {
+          setNoRoutesAvailable(true)
```

Comment:
We should check if the catched error here matches the `currentRouteCallId´, because if not we can ignore it and should not overwrite a perhaps valid result of the current call.